File: src/com/anathema_roguelike/entities/characters/perks/actions/targetingstrategies/ranges/VeryShortRange.java
package com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.ranges;

import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.Targetable;

public class VeryShortRange<T extends Targetable> extends CircularRange<T> {
	
	@SafeVarargs
	public VeryShortRange(Class<T> targetType, TargetConstraint<T, Character> ...constraints) {
		super(targetType, constraints);
	}

	@Override
	protected double getRadius(Character character) {
		return 1.0;
	}
}

File: src/com/anathema_roguelike/entities/characters/perks/actions/targetingstrategies/ranges/VeryShortRange.scala


package com.anathema_roguelike
package entities.characters.perks.actions.targetingstrategies.ranges

import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.Targetable
import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.constraints.TargetConstraint
import com.anathema_roguelike.entities.characters.Character

import scala.reflect.ClassTag
import scala.reflect.runtime.universe._

class VeryShortRange[T <: Targetable : TypeTag : ClassTag](constraints: TargetConstraint[T, Character]*) extends CircularRange[T] (constraints:_*) {

  override protected def getRadius (character: Character): Double = 1.0
}
